src/molecules/side-menu.module.css

```
@@ -0,0 +1,92 @@
+.sideMenu {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 22.5rem;
+    height: 100%;
+    background-color: var(--surface-color);
+    transform: translateX(-100%);
+    transition: transform 0.3s ease-in-out;
+    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+    display: flex;
+    flex-direction: column;
+    border-top-right-radius: 1rem;
+    border-bottom-right-radius: 1rem;
+    padding-top: 1rem;
+  }
+  
+.sideMenu[data-open="true"] {
+    transform: translateX(0);
+}
+.sideMenu[data-open="false"] {
+  transform: translateX(-100%);
+}
+  .overlay {
+    position: fixed;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background-color: rgba(0, 0, 0, 0.2);
```

```suggestion
    opacity: 0.2;
```
グローバルな CSS の `scrim` というスタイルが, このようなオーバーレイする背景の色を適用するためのものになります.
ただしこのスタイルには不透明度が設定されていないので, `side-menu.tsx` での `className` を修正したあと, さらにここで代わりに `opacity` を設定することになると思います.